enterprise/reporting/event_stats.go

```diff
@@ -0,0 +1,56 @@
+package reporting
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+const (
+	EventsDeliveredMetricName = "events_delivered_total"
+	EventsAbortedMetricName   = "events_aborted_total"
+)
+
+type EventStatsReporter struct {
+	stats            stats.Stats
+	configSubscriber *configSubscriber
+}
+
+func NewEventStatsReporter(configSubscriber *configSubscriber, stats stats.Stats) *EventStatsReporter {
+	return &EventStatsReporter{
+		stats:            stats,
+		configSubscriber: configSubscriber,
+	}
+}
+
+func (es *EventStatsReporter) Report(metrics []*types.PUReportedMetric, _ *sql.Tx) error {
```

Comment:
We are introducing transaction listeners in #3994. Stats should be recorded only after the transaction has succeeded